### PR TITLE
Add Amazon SSM extension

### PR DESCRIPTION
**Pull request by emattheis**
Just like the title says, I created an extension for exposing the Amazon SSM SDK following the existing Amazon KMS extensions as a blueprint. I added a minimal test case and have done some integration in my own products to verify functionality, but there is no new functionality to speak of - it follows the same pattern as all the uniform Amazon clients.

This API is especially useful for those of us looking to store/retrieve dynamic parameters (including encrypted secrets) using the parameter store, and  my testing has focused exclusively on that aspect of the SSM client.

### CODE REVIEW

**Comment by emattheis** [comment]
I created a quickstart for the extension as well: https://github.com/quarkusio/quarkus-quickstarts/pull/849

**Comment by emattheis** [comment]
I've created a guide for the extension as well

**Review by oscerd — APPROVED** [review]

**Comment by emattheis** [comment]
Thanks for the approval @oscerd ! What's the process for getting this merged?  I rebased against `main` and the checks were [all successful on my fork](https://github.com/emattheis/quarkus/actions/runs/806852974), but the JDK 11 Windows check still fails on the PR...

**Comment by oscerd** [comment]
This is a question for the maintainers. I worked on the iam extension and i just wanted to vote up your work by approving :-)

File: docs/src/main/asciidoc/amazon-ssm.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/amazon-ssm.adoc:12]:
```suggestion
One of the most useful features of SSM for microservices is the https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-parameter-store.html[Parameter Store], which provides secure, hierarchical storage for configuration data management and secrets management..
```

File: docs/src/main/asciidoc/amazon-ssm.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/amazon-ssm.adoc:88]:
```suggestion
In this example, we will create an application that allows us to store and retrieve parameters to and from SSM parameter store using a RESTful API.
```

**Review by gsmet — CHANGES_REQUESTED** [review]
This looks very good, thanks (and sorry for the delay!).

I added some suggestions for the documentation. I'm not sure about the callouts but could you check the generated output to make sure things are OK (the numbers should appear in small circles)?

File: docs/src/main/asciidoc/amazon-ssm.adoc
Comment (emattheis) [reply docs/src/main/asciidoc/amazon-ssm.adoc:157]: cool, I didn't know you could use that notation outside the code blocks, but it does look better that way 👍

**Comment by emattheis** [comment]
Thanks @gsmet! I update the PR with your suggested changes and rebased against main.

**Review by gsmet — APPROVED** [review]
Cool, thanks. I'll wait for CI and merge!

**Comment by gsmet** [comment]
Merged, thanks!

**Pull request merged**
